src/components/dialogs/restore-modification-dialog.js

```diff
@@ -0,0 +1,194 @@
+        flexDirection: 'column',
+        flexGrow: 1,
+        paddingBottom: theme.spacing(8),
+    }),
+    list: (theme) => ({
+        paddingTop: theme.spacing(0),
+        flexGrow: 1,
+    }),
+    modificationsTitle: (theme) => ({
+        display: 'flex',
+        alignItems: 'center',
+        margin: theme.spacing(0),
+        padding: theme.spacing(1),
+        backgroundColor: theme.palette.primary.main,
+        color: theme.palette.primary.contrastText,
+        overflow: 'hidden',
+    }),
+    toolbar: (theme) => ({
+        padding: theme.spacing(0),
+        border: theme.spacing(1),
+        minHeight: 0,
+        margin: 0,
+        flexShrink: 0,
+    }),
+    toolbarIcon: (theme) => ({
+        marginRight: theme.spacing(1),
+    }),
+    toolbarCheckbox: (theme) => ({
+        marginLeft: theme.spacing(1.5),
+    }),
+    filler: (theme) => ({
+        flexGrow: 1,
+    }),
+    dividerTool: (theme) => ({
+        background: theme.palette.primary.main,
+    }),
+    circularProgress: (theme) => ({
+        marginRight: theme.spacing(2),
+        color: theme.palette.primary.contrastText,
+    }),
+    formattedMessageProgress: (theme) => ({
+        marginTop: theme.spacing(2),
+    }),
+    notification: (theme) => ({
+        flex: 1,
+        alignContent: 'center',
+        justifyContent: 'center',
+        marginTop: theme.spacing(4),
+        textAlign: 'center',
+        color: theme.palette.primary.main,
+    }),
+    icon: (theme) => ({
+        width: theme.spacing(3),
+    }),
+};
+
+/**
+ * Dialog to select network modification to create
+ * @param {Boolean} open Is the dialog open ?
+ * @param {EventListener} onClose Event to close the dialog
+ * @param onOpenDialog handle the opening of dialogs
+ * @param dialogs the list of dialog
+ */ const RestoreModificationDialog = ({
+    open,
+    onClose,
+    modifToRestore,
+    currentNode,
+    studyUuid,
+}) => {
+    const intl = useIntl();
+
+    useEffect(() => {
+        setModificationsToRestore(modifToRestore);
+    }, [modifToRestore]);
+
+    const handleClose = () => {
+        onClose();
+    };
+
+    const [modificationsToRestore, setModificationsToRestore] = useState([]);
+
+    const handleRestore = () => {
+        const selectedModificationsUuidToRestore = [
+            ...selectedItems.values(),
+        ].map((item) => item.uuid);
+        restoreModifications(
+            studyUuid,
+            currentNode.id,
+            selectedModificationsUuidToRestore
+        );
+        handleClose();
+        console.log(selectedModificationsUuidToRestore);
+    };
+
+    const [selectedItems, setSelectedItems] = useState(new Set());
+
+    return (
+        <Dialog
+            fullWidth
+            maxWidth="xs"
+            open={open}
+            onClose={handleClose}
+            aria-labelledby="dialog-restore-modifications"
+        >
+            <DialogTitle>
+                {intl.formatMessage({ id: 'RestoreModifications' })}
```

Comment:
The title can be given as a prop, so it can be used for both nodes and modifications

Reply:
but I don't know, I can do the changes on my side since i'm doing the node part